src/audio/AudioSystem.java

```diff
@@ -137,15 +137,23 @@ public void loadBlueprint(ArrayList<String> blueprint) {
 				BlueprintLoader.getAllLinesWith("AUDIOSOURCECOMPONENT", blueprint);
 		String[] frags = null;
 		for(String dataSet : audioSourceComponentData) {
-			int eID = BlueprintLoader.extractEID(dataSet);
-			frags = BlueprintLoader.getDataFragments(dataSet);
-			attachAudioSource(eID, frags[0], 
-					Float.valueOf(frags[1]), 
-					Float.valueOf(frags[2]), 
-					Float.valueOf(frags[3]), 
-					Float.valueOf(frags[4]), 
-					Float.valueOf(frags[5]), 
-					Boolean.valueOf(frags[6]));
+			try {
+				int eID = BlueprintLoader.extractEID(dataSet);
+				if (eID < 0) {
+					continue;
+				}
+				
+				frags = BlueprintLoader.getDataFragments(dataSet);
+				attachAudioSource(eID, frags[0], 
+						Float.valueOf(frags[1]), 
+						Float.valueOf(frags[2]), 
+						Float.valueOf(frags[3]), 
+						Float.valueOf(frags[4]), 
+						Float.valueOf(frags[5]), 
+						Boolean.valueOf(frags[6]));
+			} catch (NullPointerException|IllegalArgumentException e) {
```

Comment:
Differentiate between NPE and IAE for more precise feedback. Right now there is now way to tell whether insufficient information was supplied or the information that was supplied had the wrong format.